helix-view/src/gutter.rs

```diff
@@ -139,6 +141,47 @@ pub fn diff<'doc>(
     }
 }
 
+pub fn scrollbar<'doc>(
+    _editor: &'doc Editor,
+    doc: &'doc Document,
+    view: &View,
+    theme: &Theme,
+    is_focused: bool,
+) -> GutterFn<'doc> {
+    let total_lines = doc.text().len_lines();
+    let view_height = view.inner_height();
+    let view_vertical_offset = doc.text().char_to_line(view.offset.anchor);
+
+    // Scrollbar will only be rendered on focused views where contents exceed the view height
+    if !is_focused || view_height >= total_lines {
+        return Box::new(move |_, _, _, _| None);
+    }
+
+    let height =
+        ((view_height.pow(2) as f32 / total_lines as f32).ceil() as usize).saturating_sub(1);
+
+    let scrollable_view_height = view_height.saturating_sub(height);
+    let scroll_percentage =
+        view_vertical_offset as f32 / total_lines.saturating_sub(view_height) as f32;
+
+    let start_line = view_vertical_offset
+        .saturating_add((scroll_percentage * scrollable_view_height as f32).ceil() as usize);
+
+    let style = theme.get("ui.menu.scroll");
+    Box::new(
+        move |line: usize, _selected: bool, _first_visual_line: bool, out: &mut String| {
+            let icon = if line >= start_line && line <= start_line.saturating_add(height) {
+                "█"
```

Comment:
We already use the half block character for the completion popup scrollbar, so we could use the same here for consistency:

https://github.com/helix-editor/helix/blob/15e07d4db893aec7b9e117c1f88400a68ba98ae2/helix-term/src/ui/popup.rs#L277-L281

Reply:
I originally shied away from that since there's nothing stopping someone from putting the scrollbar on the left, but it looks like the `diff` gutter is making a similar assumption and it looks better for it.

I'll add the change as suggested, but I'm open to discussing it if others have a different opinion.